### PR TITLE
Improve the way of creating menus

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes (support for options in menu builders) |
| BC breaks?      | yes (removed `Sylius\Bundle\UiBundle\Menu\AbstractMenuBuilder`) |
| Related tickets | - |
| License         | MIT |

### CODE REVIEW

**Comment by pamil** [comment]
@maggie yeah, that's quite outdated, I'll take care of it in another PR (to prevent spamming Travis)

**Comment by pjedrzejewski** [comment]
I preferred to have 1 menu builder class per menu. Otherwise they can get messy quickly. Why changing that?

**Comment by pamil** [comment]
@pjedrzejewski reverted

**Comment by pjedrzejewski** [comment]
@pamil What was the drawback of having the abstract menu builder?

File: UPGRADE.md [hunk 1]
```diff
@@ -22,10 +28,14 @@
 
 ### Mailer / MailerBundle
 
+### MoneyBundle
```
Comment (lchrusciel) [inline UPGRADE.md]: It is pretty far from this changes to the topic of this PR 🌾

File: src/Sylius/Bundle/AdminBundle/Menu/MainMenuBuilder.php [hunk 2]
```diff
@@ -11,21 +11,44 @@
...
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    /**
+     * @param array $options
+     *
      * @return ItemInterface
      */
-    public function createMenu()
+    public function createMenu(array $options)
```
Comment (lchrusciel) [inline src/Sylius/Bundle/AdminBundle/Menu/MainMenuBuilder.php]: Why did you add it? It is not used

**Comment by pjedrzejewski** [comment]
Thanks Kamil!

File: src/Sylius/Bundle/AdminBundle/Menu/MainMenuBuilder.php (hunk 2 above)
Comment (pamil) [inline src/Sylius/Bundle/AdminBundle/Menu/MainMenuBuilder.php]: It's the new feature introduced in this PR :D
